core/src/main/java/cz/xtf/core/namespace/NamespaceManager.java

```diff
@@ -58,12 +59,14 @@ public static boolean createIfDoesNotExistsProject() {
     /**
      * Creates namespace if it does not exist
      *
-     * @return true if successful, false otherwise
+     * @return true if newly created, false otherwise (failed or namespace already present)
      */
     public static boolean createIfDoesNotExistsProject(String namespace) {
         OpenShift openShift = OpenShifts.master(namespace);
 
-        boolean isCreated = false;
+        // in case namespace is terminating (means is being deleted) then wait
```

Comment:
Sometimes it happen namespace is in terminating state forever. Motivation for introducing `xtf.openshift.namespace.per.testcase=true` was to mitigate forever terminating projects. So we did not avoid our initial motivation?

Reply:
if `xtf.openshift.namespace.per.testcase=true` then it's not an issue as it's creating namespace with different name.